deployment/src/main/java/io/quarkiverse/argocd/deployment/devservices/Utils.java

```diff
@@ -0,0 +1,95 @@
+package io.quarkiverse.argocd.deployment.devservices;
+
+import java.io.InputStream;
+import java.net.URL;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import org.jboss.logging.Logger;
+
+import io.fabric8.kubernetes.api.model.Pod;
+import io.fabric8.kubernetes.client.KubernetesClient;
+
+public class Utils {
+    private static final Logger LOG = Logger.getLogger(Utils.class);
+    private static ArgocdDevServiceConfig config;
+    private static KubernetesClient client;
+
+    protected static InputStream fetchArgocdResourcesFromURL(String version) {
+        InputStream resourceAsStream = null;
+        try {
+            if (version == "latest") {
+                resourceAsStream = new URL(
+                        "https://raw.githubusercontent.com/argoproj/argo-cd/refs/heads/master/manifests/install.yaml")
+                        .openStream();
+            } else {
+                resourceAsStream = new URL(
+                        "https://raw.githubusercontent.com/argoproj/argo-cd/refs/tags/" + version + "/manifests/install.yaml")
+                        .openStream();
+            }
+        } catch (Exception e) {
+            LOG.error("The resources cannot be fetched from the argocd repository URL !");
+            LOG.error(e);
+        }
+        return resourceAsStream;
+    }
+
+    protected static InputStream fetchIngressResourcesFromURL(String version) {
+        InputStream resourceAsStream = null;
+        try {
+            if (version == "latest") {
```

Comment:
We can possibly keep that to allow users specifying `latest` as the version. I don't like it defaulting to `latest` but if user specifies `latest` we could possibly respect that. 
So, let's keep this if clause.

Reply:
I changed the default to a tagged version

Reply:
Do I have to change something else as now the default is a version ?